src/app.py

```diff
@@ -0,0 +1,90 @@
+import logging
+from fastapi import FastAPI, Request
+from fastapi.responses import JSONResponse
+
+from src.controllers.library import Library
+from src.controllers.management.borrowing_department import BorrowingDepartment
+from src.routers.borrowing_department.borrowing_department import borrowing_department_router
+from src.routers.library.library import library_router
+from utils.custom_library_errors import (
+    PatronNotFoundError,
+    LibraryItemNotFoundError,
+    LibraryItemAlreadyBorrowedError,
+    BorrowedLibraryItemNotFound,
+    BillToPayError,
+    PatronAlreadyExistsError,
+    ItemAlreadyExistsError
+)
+
+
+def create_app(library_name: str) -> FastAPI:
+    app = FastAPI()
+
+    # Initialize and store the Library instance in the app's state
+    app.state.library = Library(name=library_name)
+    app.state.borrowing_department = BorrowingDepartment()
+    # Include routers
+    app.include_router(library_router, prefix="/library")
+    app.include_router(borrowing_department_router, prefix="/borrowing")
+
+    # Set up logging
+    logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+    # Exception Handlers with Logging
+    @app.exception_handler(ItemAlreadyExistsError)
```

Comment:
Like I said, change this to catch the generic exception and return the method and status code of the specific exception